src/remediations/fifi_2.js

```diff
@@ -217,3 +219,112 @@ async function storePlaybookRun (playbook_run_id, remediation, username) {
 
     await queries.insertRHCPlaybookRun(run);
 }
+
+/**
+ * Sync dispatcher_runs table with playbook-dispatcher API for specific playbook runs
+ * 
+ * Logic:
+ * - If any dispatcher_run is failed or timeout, skip sync (no API call needed)
+ * - If pending/running dispatcher_runs exist OR no dispatcher_runs exist, sync with API
+ * - For existing runs: update status, for missing runs: create new entries
+ */
+exports.syncDispatcherRunsForPlaybookRuns = async function (playbookRunIds) {
+    if (!playbookRunIds.length) {
+        return [];
+    }
+    
+    try {
+        const sql = `
+            SELECT 
+                pr.id as playbook_run_id,
+                COUNT(dr.dispatcher_run_id)::int as total_dispatcher_runs,
+                COUNT(CASE WHEN dr.status IN ('failure', 'timeout') THEN 1 END)::int as failed_runs,
+                COUNT(CASE WHEN dr.status IN ('pending', 'running') THEN 1 END)::int as incomplete_runs
+            FROM playbook_runs pr
+            LEFT JOIN dispatcher_runs dr ON pr.id = dr.remediations_run_id
+            WHERE pr.id IN (:playbookRunIds)
+            GROUP BY pr.id
+        `;
+
+        const playbookRunCounts = await db.s.query(sql, {
+            replacements: { playbookRunIds },
+            type: db.s.QueryTypes.SELECT
+        });
+
```

Comment:
Actually, the code from this point on kinda _does_ feel like it belongs in this file.  Maybe just move the db query part to `remediations.queries.js` and leave this part with the call to playbook-dispatcher here? 🤷🏻‍♂️